rotkehlchen/db/filtering.py

```diff
@@ -102,7 +102,7 @@ def prepare(self) -> Tuple[List[str], List[Any]]:
         if self.should_join_events is True:
             query_filters.append(
                 'LEFT JOIN (SELECT event_identifier, counterparty, asset FROM history_events) '
-                'ON hex(ethereum_transactions.tx_hash)=substr(event_identifier, 3)',
+                'ON lower(hex(ethereum_transactions.tx_hash))=lower(substr(event_identifier, 3))',
```

Comment:
This fixes the problem and makes the transaction not appear if the only event it has concerns an asset that is ignored. For exampe receiving a spam token.

But this also ignores an entire transaction if one of the events it has is an ignored asset which should not be the case.

Can you:
1. Make it so that the entire transaction is ignored only if all the associated events concern ignored assets?
2. If a transaction has multiple events and some of them are of ignored assets return the transactions and the events that do not concern ignored assets
3. Write a test for this case too.

Reply:
In my opinion it is checked right [here](https://github.com/rotki/rotki/blob/develop/rotkehlchen/tests/api/test_ethereum_transactions.py#L1248).

There is a `tx1` which has 2 events. One of them is of an ignored asset. So the transaction and the other event are returned.

Isn't it what you meant?

Reply:
Just like with the previous version of the test something is off.

You can try it yourself with this PR:

1. Add an ethereum address with few transactions
2. Pick one where you are sending a token to someone
3. Add token to ignored assets
4. You will see that the entire transaction does not appear in the UI. (You shoudl expect only the burning to appear).

Reply:
That's strange..... Check the screenshot. There were 2 transactions where I was sending a DAI, then I marked DAI as an ignored asset and it disappeared but the transaction fee is there... So everything looks as expected.

What can differ in our setup? Used address, asset that was marked as an ignored, database but it shouldn't make sense. Do you have any ideas?

![image](https://user-images.githubusercontent.com/63492346/169378449-984bae11-f034-4c7c-be45-6c68543ca818.png)

Reply:
All right ... so not sure what the heck but it seems to now work fine for me too. I either misread the UI the first time when testing or rebasing your PR on develop and restarting fixed something weird.

In any case if we find anything weird here later we can fix it.